rmi/servercode/TransImpl/TransactionManager.java

```diff
@@ -121,7 +115,42 @@ public boolean commit(int id) throws InvalidTransactionException, TransactionAbo
 				}
 				if (!result)
 					return false;
+
 				System.out.println("Transaction " + transactionCounter + " Committed in Manager");
+				// For NOW:
+				// We are committing our transactions to master record
+				// once all RM's have voted yes.
+
+				// Clone master record table
+				HashVault masterClone = createNewVault();
+				masterClone.serialize_in();
+
+				// Write transaction commits to masterClone
+				System.out.println("Storing temporary transaction tables...");
+				for (int i = 0; i < toCommit.RM_Commit_Ids.size(); i++) {
+					try {
+						// Storing temporary HashVault Backup
+						masterClone.store(toCommit.RM_Commit_Ids.get(i), toCommit.activeRMs.get(i).getHash());
+						masterClone.serialize_out_temp(id);
+					}
+					catch (IOException e) {
+						e.printStackTrace();
+						return false;
+					}
+				}
+
+				// Serialize masterClone as the new Master Record
+				System.out.println("Commiting to master tables...");
+				masterClone.serialize_out();
+
+				// CURRENTLY COMMENTED OUT:
+				// This will destroy our temporary tables after we securely commit
+				// I just uncommented it for now we can debug anything to do
+				// with our shadows a bit more easily
+
+				//destroy(id);
```

Comment:
Note that this line is purposefully commented out, just for testing. If we were running the system for real we'd want to be destroying our temporary transaction tables after they 2nd round commit.